CDP4Common/Extensions/CategorizableThingExtensions.cs

```diff
@@ -55,20 +57,65 @@ public static class CategorizableThingExtensions
         /// </returns>
         public static bool IsMemberOfCategory(this ICategorizableThing categorizableThing, Category category)
         {
-            var memberOfCategories = categorizableThing.GetAllCategories();
-
             var categoriesToCheck = category.AllDerivedCategories().ToList();
+            return IsMemberOfCategory_Impl(categorizableThing, category, categoriesToCheck);
```

Comment:
being picky, but not super happy with this kind of naming... may i suggest `IsMemberOfCategoryImplementation`